From 44ecd603000b0f1caf965669e10c1ff0e1ddee49 Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Thu, 27 Jul 2017 10:23:56 +0300
Subject: [PATCH 1835/2241] fix: hwrng: omap: remove failure message when probe
 deferral occurs

Probe deferral mechanism allows drivers to report at probe time
that they cannot get all the resources required by the device,
and should be retried at a later time.
This scenario is normal and shouldn't raise an error message.

Change-Id: I1e13b788612625c0cbe796f2bbdaa1a9fdbd0039
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42274
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/char/hw_random/omap-rng.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/char/hw_random/omap-rng.c b/drivers/char/hw_random/omap-rng.c
index b21a5c7..cdfc74f 100644
--- a/drivers/char/hw_random/omap-rng.c
+++ b/drivers/char/hw_random/omap-rng.c
@@ -479,7 +479,9 @@ err_register:
 	if (!IS_ERR(priv->clk))
 		clk_disable_unprepare(priv->clk);
 err_ioremap:
-	dev_err(dev, "initialization failed.\n");
+	if (ret != -EPROBE_DEFER)
+		dev_err(dev, "initialization failed.\n");
+
 	return ret;
 }
 
-- 
2.7.4

